server.js

```diff
@@ -2,28 +2,28 @@ import http from 'http';
 import fs from 'fs';
 import path from 'path';
 
-const port = process.env.PORT || 8080;
 const root = process.env.ROOT || '.';
-const head = process.env.HEAD || null;
-
-const userHeaders = JSON.parse(head);
+const userHeaders = JSON.parse(process.env.HEAD || null);
 
 const mimeLookup = new Map([
   ['.css',  'text/css'],
   ['.html', 'text/html'],
   ['.ico',  'image/x-icon'],
   ['.js',   'application/javascript'],
   ['.json', 'application/json'],
-  ['.md', 'text/markdown'],
+  ['.md',   'text/markdown'],
   ['.png',  'image/png'],
   ['.txt',  'text/plain'],
 ]);
 
 class Server {
+
+  static origin;
+
   static requestListener(request, response) {
     if (request.method === 'GET') {
-      const fileUrl = request.url;
-      const filePath = path.resolve(`${root}/${fileUrl}`);
+      const url = new URL(request.url, Server.origin);
```

Comment:
By adopting `URL` here we can easily extract `pathname` and fix the confusion around paths which incorporate query params. Easy peasy.